### PR TITLE
[MatrixCamp] add function ltrim and rtrim

### PR DESCRIPTION
**Pull request by jiajunhuang**
**What type of PR is this?**

- [x] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue https://github.com/matrixorigin/matrixone/issues/1985
issue https://github.com/matrixorigin/matrixone/issues/1987

**What this PR does / why we need it:**

Implement `ltrim` and `rtrim`

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by broccoliSpicy — APPROVED** [review]
spectacular code...

File: pkg/builtin/unary/ltrim.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	overload.UnaryOps[builtin.Ltrim] = []*overload.UnaryOp{
+		{
+			Typ:        types.T_varchar,
+			ReturnType: types.T_varchar,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.(*types.Bytes)
+
+				// totalCount - spaceCount is the total bytes need for the ltrim-ed string
+				spaceCount := ltrim.CountSpacesFromLeft(origVecCol)
+				totalCount := int32(len(origVecCol.Data))
```
Comment (broccoliSpicy) [inline pkg/builtin/unary/ltrim.go:56]: perhaps we could reuse the original vector when possible

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

File: pkg/builtin/unary/ltrim.go (hunk 1 above)
Comment (broccoliSpicy) [reply pkg/builtin/unary/ltrim.go:56]: huh, if we reuse the original vector, it kind of like using double points to change a []types.bytes in place, then I guess we need a different implementation for this trim function, perhaps we could skip this "vector reuse procedure" to make our implementation easier

**Review by broccoliSpicy — APPROVED** [review]

**Review by domingozhang — COMMENTED** [review]
bvt passwd

**Pull request merged**